mcux/drivers/kinetis/CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ zephyr_sources_ifdef(CONFIG_HAS_MCUX_SMC       fsl_smc.c)
 zephyr_sources_ifdef(CONFIG_COUNTER_MCUX_LPTMR fsl_lptmr.c)
 zephyr_sources_ifdef(CONFIG_DAC_MCUX_DAC       fsl_dac.c)
 zephyr_sources_ifdef(CONFIG_DAC_MCUX_DAC32     fsl_dac32.c)
+zephyr_sources_ifdef(CONFIG_HAS_MCUX_TPM       fsl_tpm.c)
```

Comment:
Replace `CONFIG_HAS_MCUX_TPM` with `CONFIG_PWM_MCUX_TPM`

Reply:
Done and re-pushed. I'll redo the changes as per @galak requests in #24716 and re-push so CI is run again